flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/ExecutionFailureHandler.java

```diff
@@ -98,11 +103,22 @@ public FailureHandlingResult getGlobalFailureHandlingResult(final Throwable caus
                 true);
     }
 
+    /** @param failureListener the failure listener to be registered */
+    public void registerFailureListener(FailureListener failureListener) {
+        if (!failureListeners.contains(failureListener)) {
+            failureListeners.add(failureListener);
```

Comment:
Isn't HashSet.add() only adding something, if it isn't present already?

Reply:
Yes, no need to filter.